### PR TITLE
Issue #7:  Provide the ability to email notifications to users who re…

### LINKED ISSUE #7: Provide the ability to email notifications to users who receive them (as appropriate)
This enhancement is a new delivery channel for the Notification Hub:  email.

There should be a Quartz job that periodically emails active notifications (one time) to users designated (by a pluggable strategy) as email-based recipients.

### PR DESCRIPTION
**Pull request by drewwills**
…ceive them (as appropriate)

Resolves #7 

This pull request adds the flexible ability to send notifications as emails to some or all recipients.

@hsun1 @mairi

### CODE REVIEW

File: src/main/java/uk/ac/ed/notify/job/OutboundEmailJob.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import uk.ac.ed.notify.repository.UserNotificationAuditRepository;
+import uk.ac.ed.notify.channel.DeliveryChannelRecipientStrategy;
+import uk.ac.ed.notify.service.OutboundEmailService;
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@DisallowConcurrentExecution
```
Comment (ChristianMurphy) [inline src/main/java/uk/ac/ed/notify/job/OutboundEmailJob.java:26]: Why can't emails be sent concurrently?

**Review by ChristianMurphy — APPROVED** [review]
Looks good

**Review by mairi — APPROVED** [review]
Looks good @drewwills - nice to see this implemented :)

File: src/main/java/uk/ac/ed/notify/job/OutboundEmailJob.java (hunk 1 above)
Comment (drewwills) [reply src/main/java/uk/ac/ed/notify/job/OutboundEmailJob.java:26]: The emails can -- in fact they are, by virtue of the `@Async` annotation on `OutboundEmailService.send()`.

It's the Quartz job that figures out what emails need sending (and updates the status to say they are sent) that I don't want to run more than one of at once.

**Pull request merged**